incubator/hnc/pkg/reconcilers/object.go

```diff
@@ -548,11 +550,16 @@ func (r *ObjectReconciler) setConditions(log logr.Logger, srcInst, inst *unstruc
 		}
 
 	case err != nil:
-		// There was an error updating this object; set a local condition.
+		// There was an error updating this object; set a condition.
 		msg := fmt.Sprintf("Could not %s: %s", act, err.Error())
-		if ns.SetCondition(ao, api.CannotUpdate, msg) {
+		if !hasPropagatedLabel(inst) {
```

Comment:
Can this happen? I feel like this would be a bug, we should never try to propagate a source object.

Reply:
It should never happen. If `!hasPropagatedLabel(inst)`, it will only return `actionNop` in our current logic. And `actionNop` won't cause any error. I will remove the `hasPropagatedLabel()` check here and have a comment that "we never take actions on source objects so it should never get into this error case"

Reply:
sgtm. In the future I'd recommend doing something like `log.Error` in case something that _shouldn't_ ever happen actually _does_ (e.g. because someone else changes some logic in the future) but for now I think this is fine. Thanks!